### PR TITLE
Migrate from dep to Go modules and coreDNS 1.7

### PR DESCRIPTION
**Pull request by prameshj**
Building on top of https://github.com/kubernetes/dns/pull/384, fewer dependency changes and also no coreDNS upgrade yet.
Marking as WIP to get unit tests passing. Locally the tests fail with:

```
/tmp/go-build402259259/b518/sidecar.test flag redefined: log_dir
panic: /tmp/go-build402259259/b518/sidecar.test flag redefined: log_dir

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc0000481e0, 0x140ec20, 0x1cb4c90, 0x12855af, 0x7, 0x12afc94, 0x2f)
	/usr/local/go/src/flag/flag.go:848 +0x4ae
flag.(*FlagSet).StringVar(...)
	/usr/local/go/src/flag/flag.go:751
k8s.io/klog.InitFlags(0xc0000481e0)
	/go/pkg/mod/k8s.io/klog@v1.0.0/klog.go:420 +0x7f
k8s.io/component-base/logs.init.0()
	/go/pkg/mod/k8s.io/component-base@v0.0.0-20190620085130-185d68e6e6ea/logs/logs.go:35 +0x2d
```

I suspect this is due to the GOPATH setting using gvm.

### CODE REVIEW

**Comment by prameshj** [comment]
Fixed the sidecar unit test issue by moving to klog everywhere. Mixing use of glog and klog ends up re-defining some of the flags.

This PR is now ready for review. I verified that this builds correctly and the images work.

/assign @MrHohn 


```
./image-checks.sh 1.15.15-6-gd6bd11b-dirty gcr.io/pavithrar-k8s-dev

Verifying that iptables exists in node-cache image

Unable to find image 'gcr.io/pavithrar-k8s-dev/k8s-dns-node-cache:1.15.15-6-gd6bd11b-dirty' locally
1.15.15-6-gd6bd11b-dirty: Pulling from pavithrar-k8s-dev/k8s-dns-node-cache
Digest: sha256:5c78c95b0ee7c55409567546d39f3207f0230e78c6d6925b7ba1c2a3ca40e178
Status: Downloaded newer image for gcr.io/pavithrar-k8s-dev/k8s-dns-node-cache:1.15.15-6-gd6bd11b-dirty
iptables v1.8.2 (legacy): no command specified
Try `iptables -h' or 'iptables --help' for more information.
Verifying that node-cache binary exists in node-cache image
2020/10/05 15:29:28 [INFO] Starting node-cache image: 1.15.15-6-gd6bd11b-dirty
2020/10/05 15:29:28 [FATAL] Error parsing flags - Invalid localip specified - "", Exiting


Verifying dnsmasq-nanny startup

Unable to find image 'gcr.io/pavithrar-k8s-dev/k8s-dns-dnsmasq-nanny:1.15.15-6-gd6bd11b-dirty' locally
1.15.15-6-gd6bd11b-dirty: Pulling from pavithrar-k8s-dev/k8s-dns-dnsmasq-nanny
Digest: sha256:dc3fe01c18128b476259a6ec9ed022af7e48cd8d0c62acbad3748ac91015e303
Status: Downloaded newer image for gcr.io/pavithrar-k8s-dev/k8s-dns-dnsmasq-nanny:1.15.15-6-gd6bd11b-dirty
I1005 15:29:30.351790       1 main.go:78] opts: {{/usr/sbin/dnsmasq [] false} /etc/k8s/dns/dnsmasq-nanny 10000000000 127.0.0.1:10053}
I1005 15:29:30.356895       1 nanny.go:124] Starting dnsmasq []
W1005 15:29:30.823658       1 nanny.go:150] Got EOF from stdout
W1005 15:29:30.823664       1 nanny.go:150] Got EOF from stderr
F1005 15:29:30.823729       1 nanny.go:220] dnsmasq exited: <nil>
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc00000e001, 0xc000374000, 0x42, 0x95)
	/go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:972 +0xb8
k8s.io/klog/v2.(*loggingT).output(0x1b3e860, 0xc000000003, 0x0, 0x0, 0xc000368000, 0x1ac4724, 0x8, 0xdc, 0x0)
	/go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:921 +0x19d


Verifying kube-dns startup

Unable to find image 'gcr.io/pavithrar-k8s-dev/k8s-dns-kube-dns:1.15.15-6-gd6bd11b-dirty' locally
1.15.15-6-gd6bd11b-dirty: Pulling from pavithrar-k8s-dev/k8s-dns-kube-dns
Digest: sha256:718fe69ff59075fc467529879e297e23da0d81fe9fcd1cf364451d1652fac90d
Status: Downloaded newer image for gcr.io/pavithrar-k8s-dev/k8s-dns-kube-dns:1.15.15-6-gd6bd11b-dirty
I1005 15:29:33.002935       1 dns.go:47] version: 1.15.15-6-gd6bd11b-dirty
F1005 15:29:33.002980       1 server.go:61] Failed to create a kubernetes client: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc00000e001, 0xc0003f2000, 0xbf, 0x1d4)
	/go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:972 +0xb8
k8s.io/klog/v2.(*loggingT).output(0x2046880, 0xc000000003, 0x0, 0x0, 0xc000163e30, 0x1fb6375, 0x9, 0x3d, 0x0)
	/go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:921 +0x19d
k8s.io/klog/v2.(*loggingT).printf(0x2046880, 0x3, 0x0, 0x0, 0x14d65cf, 0x28, 0xc0001efe38, 0x1, 0x1)
	/go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:733 +0x17a


Verifying sidecar startup

Unable to find image 'gcr.io/pavithrar-k8s-dev/k8s-dns-sidecar:1.15.15-6-gd6bd11b-dirty' locally
1.15.15-6-gd6bd11b-dirty: Pulling from pavithrar-k8s-dev/k8s-dns-sidecar
Digest: sha256:7dadeee878b700256409a776d130722276825e76f61264fb0724f1af32ad022e
Status: Downloaded newer image for gcr.io/pavithrar-k8s-dev/k8s-dns-sidecar:1.15.15-6-gd6bd11b-dirty
I1005 15:29:35.229878       1 main.go:51] Version v1.15.15-6-gd6bd11b-dirty
I1005 15:29:35.229913       1 server.go:46] Starting server (options {DnsMasqPort:53 DnsMasqAddr:127.0.0.1 DnsMasqPollIntervalMs:5000 Probes:[] PrometheusAddr:0.0.0.0 PrometheusPort:10054 PrometheusPath:/metrics PrometheusNamespace:kubedns})
W1005 15:29:35.230081       1 server.go:65] Error getting metrics from dnsmasq: read udp 127.0.0.1:32976->127.0.0.1:53: read: connection refused
W1005 15:29:40.230470       1 server.go:65] Error getting metrics from dnsmasq: read udp 127.0.0.1:60944->127.0.0.1:53: read: connection refused

```

File: rules.mk [hunk 1]
```diff
@@ -171,6 +178,8 @@ test: build-dirs images-test
 	    -v $$(pwd):/go/src/$(PKG)                                          \
 	    -v $$(pwd)/bin/$(ARCH):/go/bin                                     \
 	    -v $$(pwd)/.go/std/$(ARCH):/usr/local/go/pkg/linux_$(ARCH)_static  \
+	    -v $(GOCACHE):$(GOCACHE)            	                       \
+           -e GOCACHE=$(GOCACHE)                                               \
```
Comment (MrHohn) [inline rules.mk:182]: remove one space?

**Review by MrHohn — APPROVED** [review]
Thanks! LGTM with a nit

**Review by MrHohn — APPROVED** [review]
/lgtm

**Comment by prameshj** [comment]
This failed travis build - https://travis-ci.org/github/kubernetes/dns/jobs/733123283

I will investigate.
 Image build has passed - https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/dns-push-images/1313239133279227904

**Pull request merged**